### PR TITLE
LU improvements for R8

### PR DESCRIPTION
**Pull request by vishwacsena**
- [x] Add support to pull answers from .qna file via reference
- [x] Add support to pull in alterations from .qna file via reference
- [x] Add support to pull in all utterances across all intents from a .lu file via file reference
- [x] Support round tripping LUIS settings via the .lu file format
- [x] Validations for intent, utterances, patterns, entities, phrase list limits set by LUIS

### CODE REVIEW

File: packages/lu/src/parser/luis/luisValidator.js [hunk 1]
```diff
@@ -28,9 +27,144 @@ const validateLUIS = function(luisJSON) {
...
+
+    // regex entity - 500 character max. per regular expression entity pattern
+    luisJSON.regex_entities.forEach(utt => {
+        if (utt.regexPattern.length > retCode.boundaryLimits.MAX_CHAR_REGEX_ENTITY_PATTERN) {
+            validationError(retCode.errorCode.BOUNDARY_REGEX_CHAR_LIMIT, `Regex entity '${utt.name}' with pattern /${utt.regexPattern}/ has ${utt.regexPattern.length} characters. At most ${retCode.boundaryLimits.MAX_CHAR_REGEX_ENTITY_PATTERN} is allowed.`)
+        }
+    })
+
+    // list entities: max 20000 synonyms.
+    luisJSON.closedLists.forEach(listEntity => {
```
Comment (tsuwandy) [inline packages/lu/src/parser/luis/luisValidator.js]: There is also a limit of 50 closed list. https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-boundaries

File: packages/lu/src/parser/luis/luisValidator.js [hunk 2]
```diff
@@ -28,9 +27,144 @@ const validateLUIS = function(luisJSON) {
...
+    // phrase list - Non-interchangeable phraselist has max of 5,000 phrases. 
+    totalPhrasesInApp = 0;
+    phraseLists.filter(item => !item.mode).forEach(item => totalPhrasesInApp += item.words.split(',').length);
+    if (totalPhrasesInApp > retCode.boundaryLimits.MAX_NON_INTERCHANGEABLE_PHRASES) {
+        validationError(retCode.errorCode.BOUNDARY_NINTC_PHRASES_LIMIT, `${totalPhrasesInApp} phrases found across all non-interchangeable phrase list definitions. At most ${retCode.boundaryLimits.MAX_NON_INTERCHANGEABLE_PHRASES} is allowed.`)
+    }
+
+    // Roles - 10 roles per entity
+    let totalRoles = 0;
+    ["prebuiltEntities", "patternAnyEntities", "regex_entities", "closedLists", "composites", "entities"].forEach(scope => {
```
Comment (tsuwandy) [inline packages/lu/src/parser/luis/luisValidator.js]: Nicee!

Comment (vishwacsena) [reply packages/lu/src/parser/luis/luisValidator.js]: 😎

File: packages/lu/src/parser/luis/luisValidator.js (hunk 1 above)
Comment (vishwacsena) [reply packages/lu/src/parser/luis/luisValidator.js]: added a test for this.

**Pull request merged**